board_report

```diff
@@ -11,81 +10,100 @@ source $OPENLAB_TOPDIR/Include/board_serial_op.sh
     done
 }
 
 ############## Check how many people are there in each organization ###############
 function org()
 {
-    a=`grep -Po "^([^#]*)" $OPENLAB_CONF_DIR/$USER_INFO_FILE | grep -Po "(?<=org=)[^,]+" | awk '{a[$0]++} END {for(i in a) printf("%-40s: appears ->%-4s \n"),i, a[i]}' | sort`
+    a=$(grep -v "^#" $OPENLAB_CONF_DIR/$USER_INFO_FILE | grep -Po "(?<=org=\")[^\"]+(?=\")" | awk '{a[$0]++} END {for(i in a) printf("%-40s: appears ->%-4s \n"),i, a[i]}' | sort)
     echo "$a"
 }
 
 ############## Check how many people are there in each area #######################
 function area()
 {
-    a=`grep -Po "^([^#]*)" $OPENLAB_CONF_DIR/$USER_INFO_FILE | grep -Po "(?<=area=\")([^\"]*)(?=\")" | awk '{a[$0]++} END {for(i in a) printf("%-16s: appears ->%-4s \n"),i, a[i]}' | sort`
+    a=$(grep -v "^#" $OPENLAB_CONF_DIR/$USER_INFO_FILE | grep -Po "(?<=area=\")([^\"]*)(?=\")" | awk '{a[$0]++} END {for(i in a) printf("%-16s: appears ->%-4s \n"),i, a[i]}' | sort)
     echo "$a"
 }
+
 ############## Check how many boards per user haved ###############################
 function boards()
 {
     echo "boards no: 100 is 1610(1) boards, 101 is expander boards,102 is 1610(2) boards"
     echo "130-135 is 1612_boards(D03 board)  104 is overdrive board  18,19,20 is D01 board  others boars is D02 board"
     echo "all usrs possessed boards details as following :"
     printf "%10s: -> Boards list \n" "Board usrs"
-    for value in ${board_all_usr[*]}
+
+    for usr in ${board_all_usr[*]}
     do
-        board_lists=$(grep -E "^(${value})" $OPENLAB_CONF_DIR/$USER_INFO_FILE | grep -Po "(?<=boards=)[^,]*"|tr ' ' ";")
-        printf "%-14s -> %s \n"  "${value}" "${board_lists}"
+        board_lists=$(get_user_boards $usr)
+        board_lists=$(echo "${board_lists}" | tr ' ' ',')
+        printf "%-14s -> %s \n"  "${usr}" "${board_lists}"
     done
 }
 
 ############## See the help of ./board_report.sh###############################
-function help()
+function usage()
 {
-    echo "execute \"board_report busy\" to check the current usage of all the boards"
-    echo "execute \"board_report boards\" to check how many boards per user haved"
-    echo "execute \"board_report usr\" to check all the boards are owned by which user"
-    echo "execute \"board_report area\" to check how many people are there in each area"
-    echo "execute \"board_report org\" to　check how many people are there in each organization"
+cat <<-EOF
+Usage:
+    execute "board_report status" to check the current usage of all the boards
+    execute "board_report boards" to check how many boards per user haved
+    execute "board_report owner" to check all the boards are owned by which user
+    execute "board_report area" to check how many people are there in each area
+    execute "board_report org" to　check how many people are there in each organization
+EOF
 }
 
-case $command in
-    busy)
-        busy
+if [ $# -lt 1 ]; then
+    usage
+    exit 0
+fi
+
+case $1 in
+    status)
```

Comment:
The names of local functions like status()/owner()/org()/area()/boards() are too common, and easy to conflict with other scripts' function (considering this script included many other script files).